demisto_sdk/commands/validate/validators/PA_validators/PA114_pack_metadata_version_should_be_raised.py

```diff
@@ -0,0 +1,184 @@
+    IncidentField,
+    IncidentType,
+    IndicatorField,
+    IndicatorType,
+    Job,
+    Layout,
+    LayoutRule,
+    Mapper,
+    ModelingRule,
+    Pack,
+    ParsingRule,
+    Playbook,
+    PreProcessRule,
+    Report,
+    Script,
+    Trigger,
+    Widget,
+    Wizard,
+    XDRCTemplate,
+    XSIAMDashboard,
+    XSIAMReport,
+    List,
+]
+
+
+class PackMetadataVersionShouldBeRaisedValidator(BaseValidator[ContentTypes]):
+    error_code = "PA114"
+    description = "Ensure that the pack metadata version is raised on relevant changes."
+    rationale = (
+        "When updating a pack, its version needs to be raised to maintain traceability."
+    )
+    error_message = (
+        "The pack version (currently: {old_version}) needs to be raised - "
+        "make sure you are merged from master and "
+        'update the "currentVersion" field in the '
+        "pack_metadata.json or in case release notes are required run:\n"
+        "`demisto-sdk update-release-notes -g` - for automatically generation of release notes and version\n"
+        "`demisto-sdk update-release-notes -i Packs/{pack} -u "
+        "(major|minor|revision|documentation)` for a specific pack and version."
+    )
+
+    @staticmethod
+    def should_bump_is_metadata(content_item: ContentTypes):
+        """Return if the version should be bumped as a result of the input content item.
+        Args:
+            content_item (ContentTypes): The content item for which to decide.
+
+        Return:
+            ((Boolean) True if should bump due to change False otherwise,
+             (Boolean) True if the content item is the pack_metadata False otherwise)
+        """
+        content_item_type = find_type(str(content_item.path))  # type: ignore[union-attr]
+        is_metadata = False
+
+        if (
+            not content_item_type
+            and content_item.content_type.value == Pack.content_type.value  # type: ignore[union-attr]
+        ):
+            content_item_type = FileType.METADATA
+            is_metadata = True
+
+        if content_item_type not in SKIP_RELEASE_NOTES_FOR_TYPES:
+            if content_item_type != FileType.METADATA:
+                return True, False
+            else:
+                is_metadata = True
+                old_dict = content_item.old_base_content_object.to_dict()  # type: ignore[union-attr]
+                current_dict = content_item.to_dict()  # type: ignore[union-attr]
+                for field in PACK_METADATA_REQUIRE_RN_FIELDS:
+                    if old_dict.get(field) != current_dict.get(field):
+                        return True, True
+        return False, is_metadata
+
+    def obtain_invalid_content_items(
+        self, content_items: Iterable[ContentTypes]
+    ) -> typing.List[ValidationResult]:
+        validation_results = []
+        content_packs = {}
+        content_packs_ids_to_bump = []
+        # Go over all the content items
+        for content_item in content_items:
```

Comment:
can we do it in a way that group items by pack and then check once for each pack? the content items are hashables so you should be able to do it with a dictionary
Pack Object: Changed pack items set

Reply:
wouldn't that just result in an extra for loop to sort them? why would we want this?